front/src/app/scenes/transaction/transaction.component.ts

```diff
@@ -74,85 +79,235 @@ export class TransactionComponent implements OnInit, OnDestroy {
-            let link: string;
-            if (input.name === 'tokenId') {
-              link = `/token/${processedLog.contract_address}/asset/${decodedLog[input.name]}`;
-            } else if (input.type === 'address') {
-              link = `/addr/${decodedLog[input.name]}`;
-            }
-            return {
-              link,
-              name: input.name,
-              value: decodedLog[input.name],
-            };
-          });
-          processedLog.data.push({
-            title: abiItem.name,
-            items,
-          });
-        } else {
-          const items: ProcessedLogItem[] = log.topics.map(topic => ({
-              value: <string>topic,
-          }));
-          if (log.topics.length) {
+          if (decodedLog) {
+            const items: ProcessedABIItem[] = abiItem.inputs.map(this.processAbiItem(decodedLog, address));
             processedLog.data.push({
-              title: 'topics',
+              title: abiItem.name,
               items,
             });
+          } else {
+            const items: ProcessedABIItem[] = log.topics.map(topic => (<ProcessedABIItem>{value: <string>topic}));
+            if (log.topics.length) {
+              processedLog.data.push(<ProcessedABIData>{
+                title: 'topics',
+                items,
+              });
+            }
+            if (log.data && log.data !== '0x' && log.data !== '0X') {
+              processedLog.data.push(<ProcessedABIData>{
+                title: 'data',
+                items: [{value: log.data}],
+              });
+            }
           }
-          if (log.data) {
-            processedLog.data.push({
-              title: 'data',
-              items: [{value: log.data}],
-            });
-          }
+          return processedLog;
+        });
+      }));
+  }
+
+  private processAbiItem(input: AbiInput, decoded: object, address: Address): ProcessedABIItem {
+    const item = new ProcessedABIItem();
+    item.name = input.name;
+    item.value = decoded[input.name];
+    if (address.erc_types.includes(ErcName.Go20)) {
+      if (input.name === 'value') {
+        if (address.decimals) {
+          const val = convertWithDecimals(decoded[input.name], false, true, address.decimals);
+          item.value = `${val} ${address.token_symbol}`;
         }
-        return processedLog;
-      });
-    });
+      }
+    }
+    if (address.erc_types.includes(ErcName.Go721)) {
+      if (input.name === 'tokenId') {
+        item.link = `/token/${address.address}/asset/${decoded[input.name]}`;
+        return item;
+      } else if (input.name === 'tokenURI') {
+        item.link = decoded[input.name];
+        item.linkExternal = true;
+        return item;
+      }
+    }
+    if (input.name === 'cid' && input.type === 'bytes' && !input.indexed) {
+      const hex = decoded[input.name];
+      if (hex.length > 2) {
+        try {
+          const cid: CID = new CID(Buffer.from(hex.slice(2), 'hex'));
```

Comment:
seems like second param need to be int

Reply:
actually ide was showing me that second param is biteOffset but it encoding so it's good